Gems/Multiplayer/Code/Include/Multiplayer/Components/SimplePlayerSpawnerComponentBus.h

```diff
@@ -0,0 +1,56 @@
+#include <AzCore/Outcome/Outcome.h>
+#include <AzCore/std/string/string.h>
+
+namespace Multiplayer
+{
+    //! @class SimplePlayerSpawnerRequests
+    //! @brief The SimplePlayerSpawnerRequest event-bus exposes helper methods regarding network player spawners.
+    //! Although the Multiplayer System automatically spawns in players, it's common for game specific server logic to retrieve valid spawn locations when respawning a player.
+    class SimplePlayerSpawnerRequests : public AZ::EBusTraits
+    {
+    public:
+        static const AZ::EBusHandlerPolicy HandlerPolicy = AZ::EBusHandlerPolicy::Single;
+        static const AZ::EBusAddressPolicy AddressPolicy = AZ::EBusAddressPolicy::Single;
+        
+        //! Returns the location where the next joining player will be spawned, and then advances the spawn point index.
+        //! The index will wrap back to 0 in a round-robin fashion so that a valid spawn point is always available.
+        //! Method is only valid if called from the multiplayer host/authority; clients are not given information regarding the spawn point index.
+        //! @return AZ::Transform The location of the next spawn point
+        virtual AZ::Transform RoundRobinNextSpawnPoint() = 0;
+
+        //! Returns the location where the next joining player will be spawned. Unlike RoundRobinNextSpawnPoint(), this will not cause the current spawn point index to increment.
+        //! Method is only valid if called from the multiplayer host/authority; clients are not given information regarding the spawn point index.
+        //! @return AZ::Transform The location of the next spawn point
+        virtual AZ::Transform GetNextSpawnPoint() const = 0;
+
+
+        //! Returns a mutable list of all the spawn points.
+        //! Only access/edit this list on the multiplayer host; spawn points are not synced across the network and the host is responsible for spawning players.
+        //! @return AZStd::vector<AZ::EntityId> List of spawn points.
+        virtual AZStd::vector<AZ::EntityId>& GetSpawnPoints() = 0;
+
+        //! Returns the number of spawn points.
+        //! @return Number of spawn points.
+        virtual uint32_t GetSpawnPointCount() const = 0;
+
+        //! Returns the spawn point index where the next joining player will be spawned.
+        //! @return AZ::Outcome<uint32_t, AZStd::string> If success, the value present a valid index that can be used to lookup into the spawn points array.
+        virtual AZ::Outcome<uint32_t, AZStd::string> GetNextSpawnPointIndex() const = 0;
+
+        //! Overwrites the next joining player's spawn index. The spawn index provided must be a valid (in-bounds) index into the array of available spawn points.
+        //! @return AZ::Outcome<void, AZStd::string> Success means a valid index was provided and the next spawn point was updated; failure signifies a bad (out-of-bounds) index.
+        virtual AZ::Outcome<void, AZStd::string> SetNextSpawnPointIndex(uint32_t index) = 0;
```

Comment:
Suggestion: IMO, "Outcome" is a pretty heavy-handed choice for a runtime API, especially since it's constructing strings for failure conditions in this particular case. If success/failure needs to be conveyed, maybe consider an optional or just a bool, but I'd question the overall need of conveying a result on operations as simple as these.

Alternatively, if you're committed to providing information on success/failure, then I'd question why only these two APIs get that, and not _all_ of them...

Reply:
Only Get/Set index warrant an outcome since they are the only ones that can fail. For example, GetIndex might return an out of bounds index if the array is empty (using the index in a lookup would crash), where as even if the array is empty GetNextSpawnPoint will return the actual position the next player will spawn (the worlds origin).
In either case, if the gut is that outcome is too heavy, I’ll remove it.

Reply:
Totally up to you if you want to leave or remove it, it's just a suggestion. It's not like these will get called a ton, it just struck me as odd when looking at it. From my point of view, if the API is being called from code only, I'd expect failures here to be asserts because it would mean I'm using the API wrong. If the API is being called from script, or we just want to make it friendlier to begin with, I can see making it handle the errors gracefully (which it already does) and just print AZ_ERROR. The "heavy-handed" question for me is that it's forcing the API user into some level of runtime error handling by needing to check for success/failure. To put it another way, I'm not sure what the API user _should_ do in failure conditions beyond what the API internally itself is already doing other than rewrite their code to use the API correctly.

Reply:
Fixed! I removed the outcomes. Yeah, ultimately I want this to be user friendly. I appreciate the suggestions.